Assets/MixedRealityToolkit-SDK/Features/UX/Scripts/BoundingBox/BoundingBox.cs

```diff
@@ -0,0 +1,1266 @@
+                    {
+                        closestDistance = currentDistance;
+                        closestCollider = cornerColliders[i];
+                    }
+                }
+            }
+
+            for (int i = 0; i < ballColliders.Count; ++i)
+            {
+                if (ballRenderers[i].enabled && ballColliders[i].bounds.IntersectRay(ray, out currentDistance))
+                {
+                    if (currentDistance < closestDistance)
+                    {
+                        closestDistance = currentDistance;
+                        closestCollider = ballColliders[i];
+                    }
+                }
+            }
+
+            distance = closestDistance;
+            return closestCollider;
+        }
+
+        private Ray GetHandleGrabbedRay()
+        {
+            Ray pointerRay = new Ray();
+
+            if (currentInputSource.Pointers.Length > 0)
+            {
+                currentInputSource.Pointers[0].TryGetPointingRay(out pointerRay);
+            }
+
+            return pointerRay;
+        }
+
+        private void Flatten()
+        {
+            switch (flattenAxis)
+            {
+                case FlattenModeType.FlattenX:
+                    flattenedHandles = new[] { 0, 4, 2, 6 };
+                    break;
+                case FlattenModeType.FlattenY:
+                    flattenedHandles = new[] { 1, 3, 5, 7 };
+                    break;
+                case FlattenModeType.FlattenZ:
+                    flattenedHandles = new[] { 9, 10, 8, 11 };
+                    break;
+            }
+
+            if (flattenedHandles != null)
+            {
+                for (int i = 0; i < flattenedHandles.Length; ++i)
+                {
+                    linkRenderers[flattenedHandles[i]].enabled = false;
+                }
+            }
+        }
+
+        private void SetHiddenHandles()
+        {
+            if (flattenedHandles != null)
+            {
+                for (int i = 0; i < flattenedHandles.Length; ++i)
+                {
+                    ballRenderers[flattenedHandles[i]].enabled = false;
+                }
+            }
+        }
+
+        private void GetCornerPositionsFromBounds(Bounds bounds, ref Vector3[] positions)
+        {
+            Vector3 center = bounds.center;
+            Vector3 extents = bounds.extents;
+            float leftEdge = center.x - extents.x;
+            float rightEdge = center.x + extents.x;
+            float bottomEdge = center.y - extents.y;
+            float topEdge = center.y + extents.y;
+            float frontEdge = center.z - extents.z;
+            float backEdge = center.z + extents.z;
+
+            if (positions == null || positions.Length != CORNER_COUNT)
+            {
+                positions = new Vector3[CORNER_COUNT];
+            }
+
+            positions[LeftBottomFront] = new Vector3(leftEdge, bottomEdge, frontEdge);
```

Comment:
This method is only called from SetBoundingBoxCollider which, in turn is only called from CreateRig (called whenever the state goes from inactive -> active. 

As the vectors are cached in the positions array, this change feels like a nice to have rather than a must have,